### PR TITLE
Treesitter update, wasm compilation and syntax highlighting

### LINKED ISSUE #1: canonical implementation of syntax highlighting
similar to this: https://github.com/cstrahan/tree-sitter-nix/tree/master/queries

I wanted to add crystal syntax-highlighting to `helix-editor`

https://github.com/helix-editor/helix/tree/d8df10f2958f70ee84b7c78cb7ea00aa195bc1c5/runtime/queries

### PR DESCRIPTION
**Pull request by hugopl**
This PR does 3 things in 3 patches:

- Update parser to newer tree-sitter.
- Fix WASM compilation.
- Add a highlight.scm file with captures that work on neovim (and probably helix, etc).

image

Fixes #1

### CODE REVIEW

**Comment by hugopl** [comment]
image

File: src/scanner.c [hunk 1]
```diff
@@ -102,7 +102,10 @@ typedef enum Token Token;
 /*
  * Helpful macros
  */
-#ifdef TREE_SITTER_INTERNAL_BUILD
+#ifdef __wasm
+#define DEBUG(...)
```
Comment (nobodywasishere) [inline src/scanner.c:106]: With the updated tree sitter, these debug calls should be replaced with calls to `lexer->log(...)` (eventually)

Comment (hugopl) [reply src/scanner.c:106]: Yes, this PR also need a test, I was so happy it worked that just sent it as is 😅

**Comment by hugopl** [comment]
I will close this, so we can everybody join forces on http://github.com/crystal-lang-tools/tree-sitter-crystal

**Pull request closed without merging**